src/client/input/header/gyrospace.h

```diff
@@ -0,0 +1,393 @@
+/*
+ * =======================================================================
+ *
+ * Gyro Space to Play - A plug-and-play Gyro Space Transformer code
+ * Version 0.8.5
+ *
+ * Provides functionality for transforming gyro inputs into Local Space,
+ * Player Space, and World Space, while handling sensitivity adjustments,
+ * gravity alignment and dynamic orientation.
+ *  
+ * Compatible with both C and C++ environments.
+ *
+ * Based on the work by Jibb Smart (JoyShockMapper, GamepadMotionHelpers,
+ * Fortnite v.19.30's Gyro Aim/Flick Stick implementation)
+ *
+ * =======================================================================
+ */
+
+ #ifndef GYROSPACE_HPP
+ #define GYROSPACE_HPP
+ 
+ #ifdef __cplusplus
+ #include <cmath>
+ #include <cstdio>
+ #include <cstdbool>
+ #include <cstdint>
+ extern "C" {
+ #else
+ #include "math.h"
+ #include "stdbool.h"
+ #include "stdint.h"
+ #endif
+ 
+// Ensure s32 and u32 are correctly defined
+#ifndef s32
+    #define s32 int32_t
+#endif
+
+#ifndef u32
+    #define u32 uint32_t
+#endif
+
+// Define boolean types
+#ifndef bool
+    #define bool s32
+#endif
+
+#ifndef ubool
+    #define ubool u32
+#endif
+
+// Define a small epsilon if not already defined
+#ifndef EPSILON
+    #define EPSILON 1e-5
+#endif
+
+// Debugging and Logging
+#ifdef ENABLE_GYROSPACE_DEBUG_LOGS
+    #ifdef __cplusplus
+        #include <iostream>
+        #define GYROSPACE_DEBUG_LOG(fmt, ...) std::cout << fmt
+    #else
+        #include <stdio.h>
+        #define GYROSPACE_DEBUG_LOG(fmt, ...) printf(fmt, ##__VA_ARGS__)
+    #endif
+#else
+    #define GYROSPACE_DEBUG_LOG(fmt, ...)
+#endif
+
+// Type Definitions
+typedef struct {
+    float x, y, z;
+} Vector3;
+
+typedef struct {
+    float m[4][4];
+} Matrix4;
+
+// Utility Functions
+
+/** Clamps a value between min and max. */
+static inline float clamp(float value, float min, float max) {
```

Comment:
Note: min and max aren't very safe variable names because windows has macros of that name in some standard header => if that happens to be included before gyrospace.h it won't compile because then e.g. `float min` is expanded to sth like `float (A_) < (B_) ? (A_) : (B_)` which of course makes no sense at all.